src/contexts/walletConnect/WalletConnectContext.tsx

```diff
@@ -0,0 +1,715 @@
+            respondToWalletConnectWithError(requestEvent, getSdkError('WC_METHOD_UNSUPPORTED'))
+        }
+      } catch (e) {
+        // TODO: Handle consolidation case
+        // TODO: Handle authentication requirement
+        Toast.show(getHumanReadableError(e, 'Error while building the transaction'))
+
+        posthog?.capture('Error', { message: 'Could not build transaction' })
+      }
+    },
+    // The `addresses` depedency causes re-rendering when any property of an Address changes, even though we only need
+    // the `hash`, the `publicKey`, and the `privateKey`. Creating a selector that extracts those 3 doesn't help.
+    // TODO: Figure out a way to avoid re-renders
+    [
+      walletConnectClient,
+      addresses,
+      posthog,
+      openSessionRequestModal,
+      respondToWalletConnect,
+      respondToWalletConnectWithError
+    ]
+  )
+
+  const onSessionProposal = useCallback(
+    async (sessionProposalEvent: SessionProposalEvent) => {
+      console.log('📣 RECEIVED EVENT PROPOSAL TO CONNECT TO A DAPP!')
+      console.log('👉 ARGS:', sessionProposalEvent)
+      console.log('⏳ WAITING FOR PROPOSAL APPROVAL OR REJECTION')
+
+      setSessionProposalEvent(sessionProposalEvent)
+      openSessionProposalModal()
+    },
+    [openSessionProposalModal]
+  )
+
+  const onSessionDelete = useCallback(
+    async (args: SignClientTypes.EventArguments['session_delete']) => {
+      console.log('📣 RECEIVED EVENT TO DISCONNECT FROM THE DAPP SESSION.')
+      console.log('👉 ARGS:', args)
+      console.log('🧹 CLEANING UP STATE.')
+
+      setSessionProposalEvent(undefined)
+      setActiveSessions(getActiveWalletConnectSessions(walletConnectClient))
+    },
+    [walletConnectClient]
+  )
+
+  const onSessionUpdate = useCallback((args: SignClientTypes.EventArguments['session_update']) => {
+    console.log('📣 RECEIVED EVENT TO UPDATE SESSION')
+    console.log('👉 ARGS:', args)
+  }, [])
+
+  const onSessionEvent = useCallback((args: SignClientTypes.EventArguments['session_event']) => {
+    console.log('📣 RECEIVED SESSION EVENT')
+    console.log('👉 ARGS:', args)
+  }, [])
+
+  const onSessionPing = useCallback((args: SignClientTypes.EventArguments['session_ping']) => {
+    console.log('📣 RECEIVED EVENT TO PING SESSION')
+    console.log('👉 ARGS:', args)
+  }, [])
+
+  const onSessionExpire = useCallback((args: SignClientTypes.EventArguments['session_expire']) => {
+    console.log('📣 RECEIVED EVENT TO EXPIRE SESSION')
+    console.log('👉 ARGS:', args)
+  }, [])
+
+  const onSessionExtend = useCallback((args: SignClientTypes.EventArguments['session_extend']) => {
+    console.log('📣 RECEIVED EVENT TO EXTEND SESSION')
+    console.log('👉 ARGS:', args)
+  }, [])
+
+  const onProposalExpire = useCallback((args: SignClientTypes.EventArguments['proposal_expire']) => {
+    console.log('📣 RECEIVED EVENT TO EXPIRE PROPOSAL')
+    console.log('👉 ARGS:', args)
+  }, [])
```

Comment:
I've added these one simply to help me figure out what the hell I am receiving from WC, these are all the events I could find in the docs.